### ISSUE TITLE
Documenting and fixing the unnumbered interfaces and EBGP IPv6 LLA session behavior

### ISSUE THREAD

**Issue by ipspace** [ipspace/netlab#381]
It looks like we got into another organically-grown mess regarding the unnumbered EBGP sessions. The only sane way out of it (that I can see) is documenting the expected behavior and fixing BGP configuration module to adhere to it. After that, it would be advisable to unify/standardize the device configuration templates.

* Step 1: document the expected behavior: done, see https://github.com/ipspace/netlab/blob/0ac95b7b474f21da43703c21e1f95cae0a884f71/docs/dev/unnumbered.md -- unless there are serious objections ("my device can also do X" does not count), I'll make sure BGP configuration module works as described in that document

* Step 2: Fix the BGP configuration module, and add feature tests to check device support for EBGP IPv6 LLA sessions

* Step 3: Fix configuration templates if needed. I will fix the Cumulus 4.x template, @jbemmel might have to fix the Nokia SR Linux one, and hopefully someone will figure out what to do with Cumulus 5.0 template.

**Comment by ipspace** [comment]
BTW, we might want to rename **unnumbered** neighbor flag to **ipv4_8950** to make it more aligned with what it really means, but let's agree on the basics first.

**Comment by jbemmel** [comment]
Now I remember why I moved away from using 'unnumbered' - things broke down when I started from 'unnumbered: True'

The cause is in https://github.com/ipspace/netlab/blob/dev/netsim/addressing.py#L67: As soon as you set the flag, any Booleans associated with 'ipv4' and 'ipv6' get removed

For the scenario I have in mind, the simplest config that I think could work:
```
addressing:
  fabric:
    # unnumbered: true # old style, don't set this
    ipv4: true
    ipv6: true
```

I would be ok with changing 'ipv4' into 'ipv4.rfc8950' to make the intent more explicit, or alternatively define  features.initial.ipv4.rfc8950 and interpret the above combination of ipv4/ipv6 as meaning "ipv4 with ipv6 next hops"

Another reason why the current 'unnumbered' semantics don't match with my thinking, is because of the dependency on loopback address families. In my setup I want to use ipv4-only loopbacks with ipv6 unnumbered ebgp; the 'unnumbered' logic requires an ipv6 loopback

**Comment by ipspace** [comment]
> Now I remember why I moved away from using 'unnumbered' - things broke down when I started from 'unnumbered: True'
> 
> The cause is in https://github.com/ipspace/netlab/blob/dev/netsim/addressing.py#L67: As soon as you set the flag, any Booleans associated with 'ipv4' and 'ipv6' get removed

That's true on the prefix/link level. Interfaces now always get **ipv4: True** and/or **ipv6: True** based on loopback AF.

> For the scenario I have in mind, the simplest config that I think could work:
> 
> ```
> addressing:
>   fabric:
>     # unnumbered: true # old style, don't set this
>     ipv4: true
>     ipv6: true
> ```

Correct, as you don't want to have IPv6 address on the loopback.

> I would be ok with changing 'ipv4' into 'ipv4.rfc8950' to make the intent more explicit, or alternatively define features.initial.ipv4.rfc8950 and interpret the above combination of ipv4/ipv6 as meaning "ipv4 with ipv6 next hops"

Will change **unnumbered** into **ipv4_rfc8950**. Can't use the dot syntax as that would overlap with **ipv4** neighbor attribute. Will do the same for a BGP feature. Please note it's a BGP feature, not an _initial_ feature, as RFC 8950 deals exclusively with BGP.

Guess we're all set. Will make the final adjustments and merge the changes.

**Comment by ipspace** [comment]
The new attributes are also implemented on FRR (79c2d566c7979cc425399e741e905bd5d68db251), Cumulus Linux 4.x (d644636952e1568f2b61a5b08b56ccc5d4c120ed) and 5.x with NVUE (df9eae0c313fd749870f00dd25f82f18c3d38956).
